lib/iris/tests/integration/netcdf/test__chunk_control.py

```diff
@@ -0,0 +1,161 @@
+    assert sigma.lazy_bounds().chunksize == (4, 2)
+
+
+def test_control_global(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_control_sigma_only(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(create_cube[1], model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 4, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (4, 2)
+
+
+def test_control_cube_var(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(create_cube[0], model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_control_multiple(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(
+        create_cube[0], model_level_number=2
+    ), CHUNK_CONTROL.set(create_cube[1], model_level_number=3):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (3,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_neg_one(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(model_level_number=-1):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 4, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (4,)
+    assert sigma.lazy_bounds().chunksize == (4, 2)
+
+
+def test_from_file(tmp_filepath, create_file_cube):
+    with CHUNK_CONTROL.from_file():
+        cube = iris.load_cube(tmp_filepath)
+    assert cube.shape == (2, 3, 4)
+    assert cube.lazy_data().chunksize == (1, 3, 4)
+
+
+def test_as_dask(tmp_filepath, create_cube):
+    message = "Mock called, rest of test unneeded"
+
+    loader.as_lazy_data = Mock(side_effect=RuntimeError(message))
+    with CHUNK_CONTROL.as_dask():
+        try:
+            iris.load_cube(tmp_filepath, create_cube[0])
+        except RuntimeError as e:
+            if str(e) == message:
+                pass
+            else:
+                raise e
```

Comment:
What happens if you don't include the `side_effect`? Does that avoid needing the error handling too?

Is this one of the cases where patching is more appropriate?

Reply:
Patching is more appropriate, working on that. I'm unfamiliar with Mock and Patch, and the latter doesn't seem to have "assert_called_with".

As to the `side_effect`, i'd have to set an output of the Mock so that it doesn't fail out on a different error. Which is doable, but since all we're checking is how the Mock is called, seems unnecessary, albeit uglier.

Reply:
Hopefully, patching has been fixed, used as a context manager. I still haven't made a decision on side_effect